### ISSUE TITLE
[Feature Request] 'Three Strikes' Rule for Problematic Encodes

### ISSUE THREAD

**Issue by Mushy-Snugglebites-badonkadonk** [Josh5/unmanic#127]
From time to time (and lately often times) I have files that will not convert with Unmanic. With these problem child video files I often times convert them with Handbrake.

It would be great if Unmanic would stop trying to convert these problematic files after three (or user defined) strikes and flag the file for review. Otherwise, Unmanic will continuously loop and flood the logs and thus ever growing log database which eventually slows down the web UI.

**Comment by Jadiction** [comment]
100% agree with this. I currently have ~300 failed conversion tasks from 1 movie that keeps failing over and over. If I could either put it in an "exclude list" or have a 3-strike rule, that'd be great.

**Comment by cbrherms** [comment]
100% agree with this also. 
Came back to unmanic after converting a whole series earlier in 2020....
Task history full of 75000+ failed tasks from these few files that refused to convert.
Took a while to clear/clean up due to the 500 view limit.

**Comment by Josh5** [comment]
I have added a fix in commit 2f2355163de6afbdcd30078e9e7c4105d9872e1b
This prevents files from being added when they have failed in the history.

Once this is merged into staging I will close this issue.

**Comment by cornuwel** [comment]
Hey, I just noticed a new issue that overlaps with this one and your solution (thanks btw) would make it worse:

Sometimes, the NVENC encoder on my GTX 1650 will crash (not sure why) and all subsequent transcodings will fail with the same error message: 

`[hevc_nvenc @ 0x56013e81e200] No capable devices found`

In that specific case, I definitely don't want all my other files to get blacklisted...

Maybe this error should just stop the worker and display an error? As we **know** all future tasks will fail until the device is reset?

Happy to provide any logs that can help you, and maybe create another Issue for that specific problem? Just tell me what you need :-)

**Comment by cornuwel** [comment]
https://github.com/Josh5/unmanic/issues/154

**Comment by Josh5** [comment]
> Hey, I just noticed a new issue that overlaps with this one and your solution (thanks btw) would make it worse:
> 
> Sometimes, the NVENC encoder on my GTX 1650 will crash (not sure why) and all subsequent transcodings will fail with the same error message:
> 
> `[hevc_nvenc @ 0x56013e81e200] No capable devices found`
> 
> In that specific case, I definitely don't want all my other files to get blacklisted...
> 
> Maybe this error should just stop the worker and display an error? As we **know** all future tasks will fail until the device is reset?
> 
> Happy to provide any logs that can help you, and maybe create another Issue for that specific problem? Just tell me what you need :-)

Under these new conditions those failed files will be ignored in all future library scans until you manually removed the failed entry from the history.

Later on I will convert this process into a proper blacklist section in the settings that you will be able to manage. But for now this is how it is.

**Comment by Josh5** [comment]
At a latter stage I will improve in the provided solution and develop a proper black list. 
For now I am closing this issue.
